internal/controllers/reconciliation/overrides_test.go

```diff
@@ -441,3 +444,98 @@ func TestOverrideHyphenatedFieldNames(t *testing.T) {
+	ctx := testutil.NewContext(t)
+	mgr := testutil.NewManager(t)
+	upstream := mgr.GetClient()
+
+	requireSSA(t, mgr)
+	registerControllers(t, mgr)
+	testutil.WithFakeExecutor(t, mgr, func(ctx context.Context, s *apiv1.Synthesizer, input *krmv1.ResourceList) (*krmv1.ResourceList, error) {
+		output := &krmv1.ResourceList{}
+		output.Items = []*unstructured.Unstructured{{
+			Object: map[string]any{
+				"apiVersion": "apps/v1",
+				"kind":       "Deployment",
+				"metadata": map[string]any{
+					"name":      "test-obj",
+					"namespace": "default",
+					"annotations": map[string]any{
+						"eno.azure.io/reconcile-interval": "10ms",
+						"eno.azure.io/overrides": `[
+						    { "path": "self.spec.template.spec.containers[name='foo'].resources.limits.cpu", "value": null, "condition": "has(self.spec.template.spec.containers[0].resources.limits.cpu) && !pathManagedByEno" },
+						    { "path": "self.spec.template.spec.containers[name='foo'].resources.requests.cpu", "value": null, "condition": "has(self.spec.template.spec.containers[0].resources.requests.cpu) && !pathManagedByEno" }
+						]`,
+					},
+				},
+				"spec": map[string]any{
+					"selector": map[string]any{
+						"matchLabels": map[string]any{
+							"foo": "bar",
+						},
+					},
+					"template": map[string]any{
+						"metadata": map[string]any{
+							"labels": map[string]any{
+								"foo": "bar",
+							},
+						},
+						"spec": map[string]any{
+							"containers": []any{
+								map[string]any{
+									"name":  "foo",
+									"image": "bar",
+									"resources": map[string]any{
+										"requests": map[string]any{
+											"cpu": "5m",
+										},
+										"limits": map[string]any{
+											"cpu": "10m",
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		}}
+		return output, nil
+	})
+
+	setupTestSubject(t, mgr)
+	mgr.Start(t)
+	_, comp := writeGenericComposition(t, upstream)
+
+	// Wait for initial reconciliation
+	testutil.Eventually(t, func() bool {
+		return upstream.Get(ctx, client.ObjectKeyFromObject(comp), comp) == nil && comp.Status.CurrentSynthesis != nil && comp.Status.CurrentSynthesis.Ready != nil
+	})
+
+	// The initial value should be populated
+	deploy := &appsv1.Deployment{}
+	deploy.Name = "test-obj"
+	deploy.Namespace = "default"
+	testutil.Eventually(t, func() bool {
+		mgr.DownstreamClient.Get(ctx, client.ObjectKeyFromObject(deploy), deploy)
+		return deploy.Spec.Template.Spec.Containers[0].Resources.Limits["cpu"].Equal(resource.MustParse("10m"))
+	})
+
+	// Simulate another client updating the resources value
+	err := retry.RetryOnConflict(testutil.Backoff, func() error {
+		err := mgr.DownstreamClient.Get(ctx, client.ObjectKeyFromObject(deploy), deploy)
+		if err != nil {
+			return err
+		}
+		deploy.Spec.Template.Spec.Containers[0].Resources.Limits["cpu"] = resource.MustParse("20m")
+		return mgr.DownstreamClient.Update(ctx, deploy)
+	})
+	require.NoError(t, err)
```

Comment:
Nit: Add a step to fetch the object with its managed fields and verify that it's no longer managed by Eno.

Reply:
I'll make a note to go back and fix this + improve the override tests more broadly. But I don't have time today and would like to get this merged to unblock folks 😬